src/bigrand.rs

```diff
@@ -39,13 +40,15 @@ impl<R: Rng + ?Sized> RandBigInt for R {
     fn gen_biguint(&mut self, bit_size: usize) -> BigUint {
         use super::big_digit::BITS;
         let (digits, rem) = bit_size.div_rem(&BITS);
-        let mut data = Vec::with_capacity(digits + 1);
-        for _ in 0..digits {
-            data.push(self.gen());
-        }
+        let mut data = vec![BigDigit::default(); digits + (rem > 0) as usize];
+        // `fill_bytes` is faster than many `gen::<u32>` calls
+        self.fill_bytes(data[..].as_byte_slice_mut());
+        // Swap bytes per the `Rng::fill` source. This might be
+        // unnecessary if reproducibility across architectures is not
+        // desired.
+        data.to_le();
```

Comment:
This is an interesting question -- what guarantees should we make about reproducibility?  Is this PR a breaking change altogether, since seeded RNGs may now produce a different `BigUint`?  Maybe that's going too far to consider this across crate versions, but if we want it across architectures, we'll also need to be careful in the future about conditional `BigDigit` sizes, for instance.  (Right now this is using the public `BigUint::new` though, which should always be `u32`-based even if the internals change.)

Reply:
This may be a value-breaking change (depending on the PRNG used), but I think will usually not be. (Rand does not specify that `fill_bytes` must output the same bits in the same order as repeated calls to `next_u32`, though I think this is the case for all current generators.)